astropy/io/misc/parquet.py

```diff
@@ -0,0 +1,307 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This package contains functions for reading and writing Parquet
+tables that are not meant to be used directly, but instead are
+available as readers/writers in `astropy.table`.  See
+:ref:`astropy:table_io` for more details.
+"""
+
+import os
+import warnings
+
+import numpy as np
+
+# NOTE: Do not import anything from astropy.table here.
+# https://github.com/astropy/astropy/issues/6604
+from astropy.utils.exceptions import AstropyUserWarning
+
+PARQUET_SIGNATURE = b'PAR1'
+
+__all__ = ['read_table_parquet', 'write_table_parquet']
+
+
+def parquet_identify(origin, filepath, fileobj, *args, **kwargs):
+
+    if fileobj is not None:
+        try:
+            pos = fileobj.tell()
+        except AttributeError:
+            return False
+
+        signature = fileobj.read(4)
+        fileobj.seek(pos)
+
+        return signature == PARQUET_SIGNATURE
+    elif filepath is not None:
+        return filepath.endswith(('.parquet', '.parq'))
+    else:
+        return False
+
+
+def read_table_parquet(input, columns=None, schema_only=False, filters=None):
+    """
+    Read a Table object from a Parquet file.
+
+    This requires `pyarrow <https://arrow.apache.org/docs/python/>`_
+    to be installed.
+
+    Parameters
+    ----------
+    input : str or file-like object
+        If a string, the filename to read the table from.
+        If a file-like object, the stream to read data.
+    columns : list [str], optional
+        Names of astropy columns to read.
+        This will automatically expand to all serialized columns if
+        necessary.
+    schema_only : bool, optional
+        Only read the schema/metadata with table information.
+    filters : list [tuple] or list [list [tuple] ] or None, optional
+        Rows which do not match the filter predicate will be removed from
+        scanned data.  See `pyarrow.parquet.read_table()` for details.
+
+        Predicates are expressed in disjunctive normal form (DNF), like
```

Comment:
I'd suggest moving this long explanation (from this point onward) out of the parameters and up into the main doc string. You can start with something like
```
The ``filters`` parameter consists of predicates that are expressed ...
```